swe_roe/ex2b.c

```diff
@@ -776,7 +785,8 @@ PetscErrorCode RDyComputeAdditionalEdgeAttributes(DM dm, RDyMesh *mesh) {
     if (r >= 0 && l >= 0) {
       edges->internal_edge_ids[mesh->num_internal_edges++] = iedge;
     } else {
-      edges->boundary_edge_ids[mesh->num_boundary_edges++] = iedge;
+      edges->boundary_edge_ids[mesh->num_boundary_edges++]     = iedge;
+      edges->boundary_edge_types[mesh->num_boundary_edges - 1] = REFLECTING_WALL;
```

Comment:
Not a big deal, but you could do the `++` increment in this line instead of the one preceding it, and then you wouldn't have to subtract 1.